src/handlers/gestures/useGesture.ts

```diff
@@ -0,0 +1,215 @@
+  }
+
+  function dropHandlers() {
+    for (const g of result.current.config) {
+      RNGestureHandlerModule.dropGestureHandler(g.handlerTag);
+
+      unregisterHandler(g.handlerTag);
+    }
+  }
+
+  function convertToHandlerTag(ref: GestureRef): number {
+    if (typeof ref === 'number') {
+      return ref;
+    } else if (ref instanceof BaseGesture) {
+      return ref.handlerTag;
+    } else {
+      let tag = ref.current?.handlerTag;
+      if (tag === undefined) {
+        tag = -1;
+      }
+      return tag;
+    }
+  }
+
+  function attachHandlers() {
+    if (!result.current.firstExecution) {
+      gestureConfig.initialize();
+    } else {
+      result.current.firstExecution = false;
+    }
+
+    for (const gst of gesture) {
+      RNGestureHandlerModule.createGestureHandler(
+        gst.handlerName,
+        gst.handlerTag,
+        filterConfig(gst.config, ALLOWED_PROPS)
+      );
+
+      registerHandler(gst.handlerTag, gst);
+
+      setImmediate(() => {
+        gestureConfig.prepare();
+
+        let requireToFail: number[] = [];
+        if (gst.config.requireToFail) {
+          requireToFail = gst.config.requireToFail
+            .map(convertToHandlerTag)
+            .filter((tag) => tag > 0);
+        }
+
+        let simultaneousWith: number[] = [];
+        if (gst.config.simultaneousWith) {
+          simultaneousWith = gst.config.simultaneousWith
+            .map(convertToHandlerTag)
+            .filter((tag) => tag > 0);
+        }
+
+        RNGestureHandlerModule.updateGestureHandler(
+          gst.handlerTag,
+          filterConfig(gst.config, ALLOWED_PROPS, {
+            simultaneousHandlers: simultaneousWith,
+            waitFor: requireToFail,
+          })
+        );
+      });
+    }
+    result.current.config = gesture;
+    result.current.callback?.();
+
+    if (result.current.animatedHandlers) {
+      result.current.animatedHandlers.value = (gesture.map(
+        (g) => g.handlers
+      ) as unknown) as HandlerCallbacks<Record<string, unknown>>[];
+    }
+  }
+
+  function updateHandlers() {
+    gestureConfig.prepare();
+
+    for (let i = 0; i < gesture.length; i++) {
+      const gst = result.current.config[i];
+
+      gesture[i].handlerTag = gst.handlerTag;
+      gesture[i].handlers.handlerTag = gst.handlerTag;
+    }
+
+    for (let i = 0; i < gesture.length; i++) {
+      const gst = result.current.config[i];
+
+      gst.config = gesture[i].config;
+      gst.handlers = gesture[i].handlers;
+      gst.handlers.handlerTag = gst.handlerTag;
+
+      let requireToFail: number[] = [];
+      if (gst.config.requireToFail) {
+        requireToFail = gst.config.requireToFail
+          .map(convertToHandlerTag)
+          .filter((tag) => tag > 0);
+      }
+
+      let simultaneousWith: number[] = [];
+      if (gst.config.simultaneousWith) {
+        simultaneousWith = gst.config.simultaneousWith
+          .map(convertToHandlerTag)
+          .filter((tag) => tag > 0);
+      }
```

Comment:
💡 

```suggestion
      const simultaneousWith: number[] =
      gst.config.simultaneousWith
          ?.map(convertToHandlerTag)
          ?.filter((tag) => tag > 0) ?? [];
```

Reply:
Done in 8602322bfdd6bcf87f63654c882c2a80814a9fbe